spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteDataFilesProcedure.java

```diff
@@ -292,6 +329,13 @@ public void testRewriteDataFilesWithInvalidInputs() {
         IllegalArgumentException.class, "Cannot parse predicates in where option: col1 = 3",
         () -> sql("CALL %s.system.rewrite_data_files(table => '%s', " +
             "where => 'col1 = 3')", catalogName, tableIdent));
+
+    // Test for z_order with invalid column name
+    AssertHelpers.assertThrows("Should reject calls with error message",
+        IllegalArgumentException.class, "Cannot find field 'col1' in which is specified in the ZOrder " +
```

Comment:
No need for "in the ZOrder" here. That just makes the error confusing and doesn't add much. It's clear from the sort order where the missing field is.

Reply:
originally it was like that. Russel suggested to add the Zorder in error message. 
I am fine both the ways. I will revert it back.